src/components/application_manager/src/commands/mobile/on_system_request_notification.cc

```diff
@@ -106,18 +106,28 @@ void OnSystemRequestNotification::AddHeader(BinaryMessage& message) const {
   LOG4CXX_AUTO_TRACE(logger_);
   const int timeout = policy::PolicyHandler::instance()->TimeoutExchange();
 
-  char size_str[24];
-
-  if (0 > sprintf(size_str, "%zu", static_cast<size_t>(message.size()))) {
-    memset(size_str, 0, sizeof(size_str));
-  }
-
+  size_t contentLength;
+  char size_str[24];  
   char timeout_str[24];
+
   if (0 > sprintf(timeout_str, "%d", timeout)) {
     memset(timeout_str, 0, sizeof(timeout_str));
   }
+
   std::string policy_table_string = std::string(message.begin(), message.end());
-  ParsePTString(policy_table_string);
+  
+  /* The Content-Length to be sent in the HTTP Request header should be 
+  calculated before additional escape characters are added to the 
+  policy table string. The mobile proxy will remove the escape 
+  characters after receiving this request. */
+  contentLength = policy_table_string.length();
+
+  ParsePTString(policy_table_string, contentLength);
```

Comment:
@JackLivio cann't you get required value (content length) by subtraction initial and result `policy_table_string` length ? In this case you dont need to pass variable to function